Neos.ContentRepository.Core/Classes/SharedModel/Workspace/WorkspaceName.php

```diff
@@ -61,30 +70,30 @@ public static function forLive(): self
      */
     public static function transliterateFromString(string $name): self
     {
-        try {
-            // Check if name already match name pattern to prevent unnecessary transliteration
+        if (self::hasValidFormat($name)) {
             return self::fromString($name);
-        } catch (\InvalidArgumentException $e) {
-            // Okay, let's transliterate
         }
 
-        $originalName = $name;
+        $originalName = strtolower($name);
 
         // Transliterate (transform 北京 to 'Bei Jing')
         $name = Transliterator::transliterate($name);
 
-        // Urlization (replace spaces with dash, special special characters)
-        $name = Transliterator::urlize($name);
-
         // Ensure only allowed characters are left
-        $name = preg_replace('/[^a-z0-9\-]/', '', $name);
+        $name = (string)preg_replace('/[^a-z0-9\-]/', '', $name);
 
-        // Make sure we don't have an empty string left.
-        if (empty($name)) {
-            $name = 'workspace-' . strtolower(md5($originalName));
+        // Ensure max length...
+        if (strlen($name) > self::MAX_LENGTH) {
+            $name = substr($name, 0, self::MAX_LENGTH);
         }
 
-        return new self($name);
```

Comment:
had that on my todo as well that we use `new self` here instead of creating a flyweight so thx